### PR TITLE
Jira mirror in refactor& jira mirror out

### PR DESCRIPTION
**Pull request by gal-berger**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30042

## Description

1.  Add outgoing mirror functionality
2. Add comments and attachments to mirror in
3. Fetch attachments and comments
4. Add outgoing mapper
5. Add new fields to the integration's incoming mapper
6. Add custom fields support.
7. Add changing Status via Transition.

## Minimum version of Demisto
- [ ] 5.0.0
- [X] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/Jira/Integrations/JiraV2/JiraV2.py [hunk 1]
```diff
@@ -680,30 +754,202 @@ def fetch_incidents(query, id_offset, fetch_by_created=None, **_):
...
+                str(dict_safe_get(comment, ['updated'], "", str))
+            ).replace(tzinfo=pytz.UTC)
+            if incident_modified_date <= comment_modified_date:
+                returned_comments.append(comment)
+        return returned_comments
+
+
+def get_incident_entries(issue, incident_modified_date, only_new=True, should_get_comments=True, should_get_attachments=True):
+    """
+    This function get for incident entries
```
Comment (yuvalbenshalom) [inline Packs/Jira/Integrations/JiraV2/JiraV2.py:818]: be more specific. gets the entries from the incident in order to mirror them out? getting the entries from the ticket to minor in?

File: Packs/Jira/Integrations/JiraV2/JiraV2.py [hunk 2]
```diff
@@ -680,30 +754,202 @@ def fetch_incidents(query, id_offset, fetch_by_created=None, **_):
...
+    if should_get_attachments:
+        attachments = demisto.get(issue, 'fields.attachment')
+        if attachments:
+            file_results = get_attachments(attachments, incident_modified_date, only_new)
+            if file_results:
+                entries['attachments'] = file_results
+    return entries
+
+
+def get_mapping_fields_command() -> GetMappingFieldsResponse:
```
Comment (yuvalbenshalom) [inline Packs/Jira/Integrations/JiraV2/JiraV2.py:842]: lets go thru this part together.

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
see my comments, I marked some parts I like you to explain to me when we will talk about it.

File: Packs/Jira/Classifiers/classifier-classifier-mapper-outgoing-Jira.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{
+ "description": "",
```
[the remark this answers is not in the document]
Comment (gal-berger) [reply Packs/Jira/Classifiers/classifier-classifier-mapper-outgoing-Jira.json:2]: Added

File: Packs/Jira/Classifiers/classifier-mapper-incoming-JiraV2.json [hunk 4]
```diff
@@ -1,28 +0,0 @@
-{
- "description": "Jira V2 mirror in classifier.",
```
[the remark this answers is not in the document]
Comment (gal-berger) [reply Packs/Jira/Classifiers/classifier-mapper-incoming-JiraV2.json:2]: you're right. I've changed the name back.

File: Packs/Jira/Integrations/JiraV2/README.md [hunk 5]
```diff
@@ -40,7 +40,10 @@ ConsumerKey__
 ## Fetched Incidents Data
 ---
 When you enable fetched incidents, Demisto fetches the first batch of Jira issues from the 10 minutes prior to when the integration was added. After the first batch of fetched issues, Demisto fetches new Jira issues as soon as they are generated in Jira. By default, 50 issues are pulled for each call. To pull older Jira issues, use the query to fetch issues option.
+If `Fetch comments` is enabled, Demisto fetches all comments for Jira issues.
```
[the remark this answers is not in the document]
Comment (gal-berger) [reply Packs/Jira/Integrations/JiraV2/README.md:43]: Fixed

File: Packs/Jira/Integrations/JiraV2/README.md [hunk 6]
```diff
@@ -40,7 +40,10 @@ ConsumerKey__
...
+If `Fetch attachments` is enabled, Demisto fetches all attachments for Jira issues.
```
[the remark this answers is not in the document]
Comment (gal-berger) [reply Packs/Jira/Integrations/JiraV2/README.md:44]: Fixed

File: Packs/Jira/Integrations/JiraV2/README.md [hunk 7]
```diff
@@ -40,7 +40,10 @@ ConsumerKey__
...
 If mirror `Mirror incoming incidents` is enabled, any incident data changed in remote JIRA server will reflected on existing fetched incidents.
+If mirror `Mirror outgoing incidents` is enabled, any incident data changed in existing fetched incidents will reflected on remote JIRA server.
```
[the remark this answers is not in the document]
Comment (gal-berger) [reply Packs/Jira/Integrations/JiraV2/README.md:46]: Fixed

File: Packs/Jira/Integrations/JiraV2/JiraV2.py [hunk 8]
```diff
@@ -303,15 +314,41 @@ def create_incident_from_ticket(issue):
         elif demisto.get(issue, 'fields.priority.name') == 'Low':
             severity = 1
 
+    file_names = []
+    if demisto.params().get('fetch_attachments'):
+        for file_result in get_entries_for_fetched_incident(issue.get('id'), False, True)['attachments']:
```
[the remark this answers is not in the document]
Comment (gal-berger) [reply Packs/Jira/Integrations/JiraV2/JiraV2.py:319]: Yes, by default the return value of `get_entries_for_fetched_incident` function is:
 entries = {'comments': [], 'attachments': []}

File: Packs/Jira/Integrations/JiraV2/JiraV2.py [hunk 9]
```diff
@@ -303,15 +314,41 @@ def create_incident_from_ticket(issue):
...
+            if file_result['Type'] != entryTypes['error']:
+                file_names.append({
+                    'path': file_result.get('FileID', ''),
+                    'name': file_result.get('File', '')
+                })
+    if demisto.params().get('fetch_comments'):
+        labels.append({'type': 'comments', 'value': str(get_entries_for_fetched_incident(issue.get('id'), True, False)
```
[the remark this answers is not in the document]
Comment (gal-berger) [reply Packs/Jira/Integrations/JiraV2/JiraV2.py:326]: Yes, by default the return value of get_entries_for_fetched_incident function is:
entries = {'comments': [], 'attachments': []}

File: Packs/Jira/Integrations/JiraV2/JiraV2.py (hunk 1 above)
Comment (gal-berger) [reply Packs/Jira/Integrations/JiraV2/JiraV2.py:818]: I use this function not only for the mirroring. but I can add the type of entries.

File: Packs/Jira/IncidentTypes/Jira_Incident.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+ "hoursR": 0,
+ "daysR": 0,
+ "weeksR": 0,
+ "system": false,
+ "readonly": false,
+ "default": false,
+ "autorun": false,
+ "disabled": false,
+ "reputationCalc": 0,
+ "layout": "bd26cd8b-f36a-4663-87be-be3537fd53aa",
```
Comment (yuvalbenshalom) [inline Packs/Jira/IncidentTypes/Jira_Incident.json:21]:
```suggestion
 "layout": "Jira Incident",
```

File: Packs/Jira/ReleaseNotes/1_3_0.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
+
+#### Classifiers
+##### classifier-mapper-incoming-JiraV2
+- Add new fields to the integration's incoming mapper.
+##### New: classifier-mapper-outgoing-Jira
+- Jira V2 mirror out classifier. (Available from Cortex XSOAR 0.0.0).
```
Comment (yuvalbenshalom) [inline Packs/Jira/ReleaseNotes/1_3_0.md:6]:
```suggestion
- Jira V2 mirror out classifier. (Available from Cortex XSOAR 6.0.0).
```

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
see my comments, mostly fixes in the incident fields/type.

this is not final, still reviewing the fetch

File: Packs/Jira/Integrations/JiraV2/JiraV2.py [hunk 12]
```diff
@@ -799,10 +799,8 @@ def get_attachments(attachments, incident_modified_date, only_new=True):
                 file_results.append(fileResult(filename=filename, data=attachments_zip))
         else:
             for attachment in attachments:
-                attachment_modified_date: datetime = parse(
-                    str(dict_safe_get(attachment, ['created'], "", str))
-                ).replace(tzinfo=pytz.UTC)
-                if incident_modified_date <= attachment_modified_date:
+                attachment_modified_date: datetime = parse(dict_safe_get(attachment, ['created'], "", str))
+                if incident_modified_date < attachment_modified_date:
```
Comment (yuvalbenshalom) [inline Packs/Jira/Integrations/JiraV2/JiraV2.py]: not sure about dose changes with removing utc.
lets talk about that and explain time.

**Review by yuvalbenshalom — COMMENTED** [review]
lets talk about the times change

**Pull request closed without merging**